scipy/stats/stats.py

```diff
@@ -4401,9 +4407,16 @@ def _perm_test(x, y, stat, compute_distance, reps=1000, workers=-1):
     null_dist : list
         The approximated null distribution.
     """
-    # parallelizes with specified workers over number of reps
+    # parallelizes with specified workers over number of reps and set seeds
     mapwrapper = MapWrapper(workers)
-    parallelp = _ParallelP(x=x, y=y, compute_distance=compute_distance)
+    if workers == 1:
+        random_state = check_random_state(0)
```

Comment:
So if workers=1, explicit `random_state` arg is ignored?

Reply:
I guess the way I am doing it where the seeds are assigned for each replication, I guess doing something different for ``workers == 1`` doesn't really matter. Fixed here https://github.com/scipy/scipy/pull/11152/commits/ebb0da401d8291008e7866c076ee717daf4e9ae0